### PR TITLE
TER-82 assert column types on typed tables

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: TER-82

check columns when it is typed table

Before asking for review make sure that:

- [x] In the case of **modified import queries**, I checked that the [CustomQuery component](https://github.com/keboola/app-custom-query-manager) is updated.

---

### CODE REVIEW

File: tests/functional/Teradata/ToStage/StageImportTest.php [hunk 1]
```diff
@@ -123,4 +127,63 @@ public function testItWontFitIn(): void
...
+
+        $source = new Table(
+            $this->getDestinationDbName(),
+            'sourceTable',
+            ['id', 'first_name', 'last_name'],
+            []
+        );
+
+        $this->expectException(ColumnsMismatchException::class);
+        $this->expectExceptionMessage('Source destination columns name mismatch. "last_name"->"middle_name"');
```
Comment (martinjunger) [inline tests/functional/Teradata/ToStage/StageImportTest.php:176]: tady kontrolujes jenom nazvy sloupcu, ale mela by tu byt i kontrola typu?

Comment (jirkasemmler) [reply tests/functional/Teradata/ToStage/StageImportTest.php:176]: ne, kontroluje se tam i typ a a delka https://github.com/keboola/php-db-import-export/blob/master/src/Backend/Assert.php#L17

Comment (martinjunger) [reply tests/functional/Teradata/ToStage/StageImportTest.php:176]: jo, to vim, ze se to tam kontroluje, ale myslel jsem spis takovejhle test: https://github.com/keboola/php-db-import-export/blob/master/tests/unit/Backend/Synapse/AssertTest.php#L465
Ale je to uz pokryty v `AssertTest`, tak cajk.

**Review by martinjunger — APPROVED** [review]
lgtm

**Pull request merged**